Add guide dynamic libraries

duianto opened the pull request.

duianto committed:
Add guide dynamic libraries

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+To switch between dynamic libraries, we add library chords in the device
+manager's library editor: https://charachorder.io/config/chords/
+
+First we add a base library chord. This exits any dynamic library.
```
Add a comma after "First"

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+#. Add an impulse chord: ``i + DUP``
+#. At the impulse output prompt, type: ``arm``
+#. Press: ``enter``
+#. At the impulse input prompt, chord: ``ar``
```
Change the chord input to our chord notation, "a+r"

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+   on a new line. We'll make it easier to erase the library name in the
+   next section.
+
+#. Add an impulse chord: ``i + DUP``
```
MODIFY
"Open the impulse chord menu:"

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+   next section.
+
+#. Add an impulse chord: ``i + DUP``
+#. At the impulse output prompt, type: ``arm``
```
ADD
"... type your desired output:"

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+
+   .. note::
+      
+      It doesn't overwrite an existing chord on the base library.
```
REVISE
"Adding a chord in a dynamic library doesn't overwrite..."

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+
+   Now, while we're on the base library: ``base``
+   
+   * when we chord: ``ar``
```
Chording notation

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+Select the dynamic library name automatically
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This isn't required, but it's a convenience.
```
REVISE
"This step isn't required..."

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,122 @@ OUTPUT, 5. CONFIRM INPUT
+When we switch between dynamic libraries, the library chord outputs the
+library name. It's an extra step to erase the library name manually.
+
+Instead, let's select the dynamic library name automatically, by adding
```
Remove the comma

CCMaverick reviewed (changes requested):
Easy to understand manual. I never took the time to learn this, but now I know how to do it lol. Minor changes

duianto committed:
chords, dyn lib, review suggestions

duianto committed:
chords, dyn lib, change examples, and update screenshots

- Replaced the two letter chord with a three letter chord
- Changed the image format from .webp to .png, because GitHubs diff doesn't support webp, with png the images are visible while reviewing a PR on GitHub.

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+
+Dynamic Libraries
+~~~~~~~~~~~~~~~~~
+
```
I'm thinking we give a brief explanation of what Dynamic Libraries are.

"Dynamic Libraries are a way with which you can store multiple different chord libraries on a device. You can switch between the different libraries by using a pre-defined chord. With Dynamic Libraries, you can use the same chord to produce different outputs, depending on what library is currently active."

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+* ``Eneot of Draconia`` in a library for Novel writing
+* ``OnePlus`` in a library for Cell Phone blogging
+
+Add dynamic library switching chords
```
Let's rename this to Dynamic Toggle Chords"
I think Dynamic toggle is pretty clear and unique enough that it won't be confused with other CC functions. You can add a note explaining this name immediately before the instructions in the section below. "We call these chords Dynamic Toggles." I kinda like your "library chords" but that can be confused with general chords in a library, which is how I was reading it until I really thought about it lol. I thought you meant to simply add a chord into the library, but then I realized that you're calling them Library chords... so, if I was confused I can imagine others may get confused as well.

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+* chord: ``b+a+l+i``
+* output: ``{action: base library}base``
+
+Then, we add a work dynamic library chord.
```
Because we're dealing with new vocabulary that we're making up, we need to be careful with being extremely clear what each word means. "work dynamic library chord". Does that mean a working library? As in, simply a Dynamic library? Or does it mean something else? I propose revising to:
"Then, we add a Dynamic Toggle for a library used at work."

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+
+.. note::
+
+   Both the chords ``b+a+l+i`` and ``w+l+i+b``, and the dynamic library
```
In the spirit of being super clear, consider:
"The chords we used in the examples above (``b+a+l+i`` and ``w+l+i+b``) and their outputs (``base`` and ``wlib``) are merely examples; they don't have to be exactly that. You can make the chord inputs and their outputs whatever you want, as long as you include the dynamic library action in the chord output."

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Switch to the work library, by chording: ``w+l+i+b``
+#. It switches to the work library and outputs its name: ``wlib``
```
ADD: "in plain text." to the end of this last sentence.

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+
+#. Switch to the work library, by chording: ``w+l+i+b``
+#. It switches to the work library and outputs its name: ``wlib``
+#. You can ignore the name, erase it or just press ``enter`` to continue
```
In the spirit of keeping readable and consistent documentation, I like making any function keys and modifiers all caps. Idk how they appear on .io, but I really thing using all caps is a useful way to differentiate between modifiers and text. Thoughts?

duianto replied:
Uppercase key names, modifiers, etc. [#116](https://github.com/CharaChorder/docs/pull/116)

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+Add a chord in a dynamic library
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Switch to the work library, by chording: ``w+l+i+b``
```
Attempting to keep consistency, all of these should either have a period at the end, or not. I don't have a preference (I think), but we should be consistent.

duianto replied:
The period was removed after the only list item with a period at the end.

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+   * when we chord: ``r+a+l``
+   * it outputs: ``realtor@email.com``
+
+#. Switch to the base library by using the base library chord we
```
REVISE:
"...using the base Dynamic Toggle that we created earlier:"

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+   * when we chord: ``r+a+l``
+   * it outputs: ``really``
+
+Select the dynamic library name automatically
```
How about we make this a dropdown?

duianto replied:
### With the last section as a dropdown.

The last section might be overlooked.
The dropdown box with a right arrow on the right side, might be seen as just another navigation button, like the previous and next buttons below it.

And if a link to the last section, is added in the previous section:
> We’ll get rid of the printed library name in the next section

Clicking on the link scrolls us to the bottom of the page, but it isn't obvious that the link is referring to the dropdown at the bottom of the page.

<img width="1278" height="944" alt="image" src="https://github.com/user-attachments/assets/6bd76dd1-631c-413a-a9ee-ec6f799fc0d6" />

### Without a dropdown.

You can't miss the there's another section when scrolling down the page.

Clicking the link: next section
Scrolls the section header: Select the dynamic library name automatically
to the top of the page.

<img width="1278" height="944" alt="image" src="https://github.com/user-attachments/assets/acc3d42d-1430-4d36-95d9-b96eb69b708f" />

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+library name. It's an extra step to erase the library name manually.
+
+Instead, let's select the dynamic library name automatically by adding
+these actions to the end of the library chord output.
```
REVISE:
"...end of the Dynamic Toggle chord output."

CCMaverick commented on docs/Chords.rst:
```diff
@@ -228,3 +228,149 @@ OUTPUT, 5. CONFIRM INPUT
+   {release_next}{left_shift}
+
+The number of ``{arrow left}`` actions should match the number of
+characters in the base and dynamic library names, in this case we used
```
REVISE:
"...in the dynamic library's name; in this case, we used..."

CCMaverick reviewed (commented):
Some comments for your consideration.

duianto committed:
chords, dyn lib, more review suggestions, etc.

Added an introduction.

Changed: Dynamic Library Switching Chords
And: Library Chords
To: Dynamic Toggles

Uppercased key names: ENTER, BACKSPACE

Added images:
Base Library Action
Dynamic Library Action
Dynamic Toggle chords without selecting the library names

duianto commented:
Ready for review
